internal/core/operations/rooms/add/executor.go

```diff
@@ -88,24 +89,54 @@ func (ex *Executor) Execute(ctx context.Context, op *operation.Operation, defini
 		amount = ex.config.AmountLimit
 	}
 
-	errGroup, errContext := errgroup.WithContext(ctx)
-	executionLogger.Info("start adding rooms", zap.Int32("amount", amount))
+	var wg sync.WaitGroup
+	errsCh := make(chan error, amount)
+
 	for i := int32(1); i <= amount; i++ {
-		errGroup.Go(func() error {
-			err := ex.createRoom(errContext, scheduler, executionLogger)
-			return err
-		})
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+			defer cancel()
+
+			err := ex.createRoom(ctx, scheduler, executionLogger)
```

Comment:
nit: haven't you changed to the below?

```suggestion
			errsCh <- ex.createRoom(ctx, scheduler, executionLogger)
```